diff --git android-project/app/src/main/java/org/libsdl/app/SDLActivity.java android-project/app/src/main/java/org/libsdl/app/SDLActivity.java
--- android-project/app/src/main/java/org/libsdl/app/SDLActivity.java
+++ android-project/app/src/main/java/org/libsdl/app/SDLActivity.java
@@ -352,7 +352,13 @@
      */
     protected int getInitSubsystems() {
         return SDL.SDL_INIT_EVERYTHING;
+    }
+
+    //---:clay:start
+    protected void preWindowCreate() {
+        // Overrided in subclass
     }
+    //---:clay:end
 
     public static void initialize() {
         // The static nature of the singleton and Android quirkyness force us to initialize everything here
@@ -476,6 +482,10 @@
                 }
             }
         }
+
+        //---:clay:start
+        preWindowCreate();
+        //---:clay:end
 
         // Set up JNI
         SDL.setupJNI(getInitSubsystems());
@@ -1097,6 +1107,29 @@
                     InputMethodManager imm = (InputMethodManager) context.getSystemService(Context.INPUT_METHOD_SERVICE);
                     imm.hideSoftInputFromWindow(mTextEdit.getWindowToken(), 0);
 
+                    //---:clay:start
+                    // Explicit request fullscreen when keyboard is hidden
+                    new Handler(android.os.Looper.getMainLooper()).post(new Runnable() {
+                        @Override
+                        public void run() {
+
+                            Activity activity = (Activity)getContext();
+                            if (activity != null) {
+                                View decorView = activity.getWindow().getDecorView();
+
+                                decorView.setSystemUiVisibility(
+                                        View.SYSTEM_UI_FLAG_LAYOUT_STABLE
+                                                | View.SYSTEM_UI_FLAG_LOW_PROFILE
+                                                | View.SYSTEM_UI_FLAG_LAYOUT_HIDE_NAVIGATION
+                                                | View.SYSTEM_UI_FLAG_LAYOUT_FULLSCREEN
+                                                | View.SYSTEM_UI_FLAG_HIDE_NAVIGATION
+                                                | View.SYSTEM_UI_FLAG_FULLSCREEN
+                                                | View.SYSTEM_UI_FLAG_IMMERSIVE_STICKY);
+                            }
+                        }
+                    });
+                    //---:clay:end
+
                     onNativeScreenKeyboardHidden();
 
                     mSurface.requestFocus();
